socorro/external/boto/crashstorage.py

```diff
@@ -28,13 +30,101 @@ def wait_time_generator():
         yield i
 
 
+class CrashidMissingDatestamp(Exception):
+    """Indicates the crash id is invalid and missing a datestamp."""
```

Comment:
We have a `socorro.lib.ooid.is_crash_id_valid`. It validates the structure, but doesn't verify the datestamp at the end is a valid datestamp. I thought about fixing that which would allow me to reduce this code a bit. I decided it was outside the scope of this change.

At some point, we should go through `socorro.lib.ooid` and similar things in Antenna and unify them into a single library that is in both projects.